scripts/init.sql

```diff
@@ -159,40 +463,140 @@ BEGIN
-        FROM available_routes(%L, %L) AS ar
-    ) AS available;', $1, $2);
+        FROM arrow.available_routes(%L, %L, %L, %L) AS ar
+    ) AS available;', $1, $2, $3, $4);
 END;
-$$ LANGUAGE plpgsql;
+$$ LANGUAGE plpgsql
+SECURITY DEFINER
+SET search_path = arrow, pg_temp;
 
-CREATE OR REPLACE FUNCTION best_path (uuid, uuid, timestamptz, timestamptz)
-RETURNS TABLE (seq integer, start_id uuid, end_id uuid, distance_meters double precision)
-AS $body$
+CREATE OR REPLACE FUNCTION arrow.best_path (
+    start_node_id INTEGER,
+    end_node_id INTEGER,
+    start_nofly_id INTEGER,
+    end_nofly_id INTEGER,
+    start_time TIMESTAMPTZ,
+    end_time TIMESTAMPTZ
+) RETURNS TABLE (
+    path_seq INTEGER,
+    start_id INTEGER,
+    end_id INTEGER,
+    distance_meters FLOAT
+) AS $body$
 BEGIN
     RETURN QUERY
         SELECT
-            results.path_seq,
-            (SELECT arrow_id FROM arrow.rnodes arn WHERE arn.id = ar.id_source),
-            (SELECT arrow_id FROM arrow.rnodes arn WHERE arn.id = ar.id_target),
-            ar.distance_meters
+            results.seq,
+            ar.id_source,
+            ar.id_target,
+            results.cost
         FROM arrow.routes ar
         JOIN (
-            SELECT path_seq, edge, cost
+            SELECT seq, edge, cost
             FROM pgr_aStar(
-                (SELECT * FROM a_star_sql_query($3, $4)),
-                (SELECT arn.id FROM arrow.rnodes AS arn WHERE arn.arrow_id = $1),
-                (SELECT arn.id FROM arrow.rnodes AS arn WHERE arn.arrow_id = $2),
+                (SELECT * FROM arrow.a_star_sql_query($3, $4, $5, $6)),
+                $1,
+                $2,
                 directed => true,
                 heuristic => 2
             )
         ) AS results ON ar.id = results.edge;
 END;
-$body$ LANGUAGE plpgsql;
+$body$ LANGUAGE plpgsql
+SECURITY DEFINER
+SET search_path = arrow, public, pg_temp;
+
+CREATE OR REPLACE FUNCTION arrow.best_path_p2p (
+    start_node UUID, -- Vertiport
+    end_node UUID, -- Vertiport
+    start_time TIMESTAMPTZ,
+    end_time TIMESTAMPTZ
+) RETURNS TABLE (
+    path_seq INTEGER,
+    start_type arrow.NodeType,
+    start_latitude FLOAT,
+    start_longitude FLOAT,
+    end_type arrow.NodeType,
+    end_latitude FLOAT,
+    end_longitude FLOAT,
+    distance_meters FLOAT
+) AS $body$
+DECLARE
+    start_node_id INTEGER;
+    start_nofly_id INTEGER;
+    end_node_id INTEGER;
+    end_nofly_id INTEGER;
+BEGIN
+    --- Get Node and Nofly IDs for start and end nodes
+    SELECT node_id, nofly_id
+        INTO start_node_id, start_nofly_id
+        FROM arrow.vertiports WHERE arrow_id = $1;
+
+    SELECT node_id, nofly_id
+        INTO end_node_id, end_nofly_id
+        FROM arrow.vertiports WHERE arrow_id = $2;
+
+    --- Get best path between nodes
+
+    RETURN QUERY
+        SELECT
+            results.path_seq,
+            (SELECT node_type FROM arrow.nodes WHERE id = results.start_id),
```

Comment:
Could use a little help here on how to structure this so I'm not making this many calls